interfaces/apparmor/backend.go

```diff
@@ -397,15 +397,16 @@ func addUpdateNSProfile(snapInfo *snap.Info, opts interfaces.ConfinementOptions,
 	policy := templatePattern.ReplaceAllStringFunc(updateNSTemplate, func(placeholder string) string {
 		switch placeholder {
 		case "###SNAP_NAME###":
-			return snapInfo.Name()
+			// TODO parallel-install: use of proper instance/store name
+			return snapInfo.InstanceName()
```

Comment:
My suggestion for this part would be to map as follows:

`###SNAP_NAME###` => `snapInfo.Name()`
`###SNAP_INSTANCE_KEY###` => `snapInfo.InstanceKey()`

And adjust the apparmor template accordingly. This doesn't have to happen in this PR obviously.

Reply:
The names should map the same names we end up using in the snap info itself, whatever these end up being, otherwise it'll be a sure bug, perhaps a serious one, sooner or later.